[rithanisk] iP

rithanisk opened the pull request:
### DukePro

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

rithanisk committed:
Implement chatbot greeting and goodbye functions

rithanisk committed:
Add Level 1

rithanisk committed:
Add Task class

rithanisk committed:
Add TaskList class

rithanisk committed:
Add getTaskListLength() method

rithanisk committed:
Add displayList() method

rithanisk committed:
Add addTask() method to add a new task to the taskList

rithanisk committed:
Add Level 2

rithanisk committed:
Add isDone() method to Task

rithanisk committed:
Add markTask method to mark a task in the task list

rithanisk committed:
Add unmarkTask to unmark a task from the task list

rithanisk committed:
Modify comments for all public methods

rithanisk committed:
Refactor code that displays newly added task

rithanisk committed:
Add Todo class

rithanisk committed:
Add Deadline class

rithanisk committed:
Add Event class

rithanisk committed:
Add toString method to Task class

rithanisk committed:
Add toString() method to subclasses of Task class

rithanisk committed:
Handle Todo task command

rithanisk committed:
Handle Deadline task command

rithanisk committed:
Modify string in displayList() method

rithanisk committed:
Handle Event task command

rithanisk committed:
Add comments for all public methods

rithanisk committed:
Change expected output from Duke's output to Nebula's output

rithanisk committed:
Update EXPECTED.TXT with expected output

rithanisk committed:
Update input.txt with input commands

rithanisk committed:
Update runtest.sh for testing

rithanisk committed:
Add NebulaException class

rithanisk committed:
Add validateCommand() method to handle incorrect user input

rithanisk committed:
Update TaskList class to store tasks in ArrayList

rithanisk committed:
Refactor text UI testing code

rithanisk committed:
Add deleteTask() method to TaskList

rithanisk committed:
Add displayDeletedTask() method to Ui class

rithanisk committed:
Add TaskType Enum

rithanisk committed:
Update Nebula class to use TaskType enum

rithanisk committed:
Add javadoc comments to Nebula and NebulaException methods

Jalenleeruixian commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+     *
+     * @param task The Task object to be added to the TaskList
+     */
+    public String addTask(Task task) {
```
Correct code naming followed

Jalenleeruixian commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+     *
+     * @return The number of tasks in the TaskList
+     */
+    public static int getTaskListLength() {
```
Good naming convention used

Jalenleeruixian commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,43 @@
+     * @param command The command string containing the command and task description
+     * @return The task description as a String
+     */
+    public String splitCommandAndTaskDescription(String command) {
```
Good naming convention followed

Jalenleeruixian commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,43 @@
+public class Parser {
```
Used noun for class. Good.

Jalenleeruixian commented on src/main/java/Nebula.java:
```diff
@@ -0,0 +1,189 @@
+     * @param taskList the current list of tasks
+     * @throws NebulaException if the command is invalid or improperly formatted
+     */
+    public static void validateCommand(String command, TaskList taskList) throws NebulaException {
```
Used verb for method. Good.

Jalenleeruixian reviewed (commented):
Overall, naming convention was adhered to.

chengzihao123 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+
+    /**
+     * Returns a string representation of the Deadline object.
+     * The string includes a "[D]" prefix to indicate that this is a Deadline task,
```
Good that you have included a javadoc for the modified toString() method even though it is not required to. According to coding standards maybe you could consider to remove the @return line and only provide additional details about your modification

chengzihao123 commented on src/main/java/Nebula.java:
```diff
@@ -0,0 +1,189 @@
+
+                TaskType taskType = parseTaskType(command);
+
+                switch (taskType) {
```
Good us of switch and case. But the indentation level of switch and case should be the same according to the coding standard that we adhere to.

chengzihao123 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+    public String markTask(int taskNum) {
+        Task t = taskList.get(taskNum - 1);
+        if (t.isDone()){
+            return ui.displayAlreadyMarkedTask();
```
good use of verb to name a function!

chengzihao123 commented on src/main/java/TaskType.java:
```diff
@@ -0,0 +1,3 @@
+public enum TaskType {
```
Good use of enumeration!

chengzihao123 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,138 @@
+import java.util.ArrayList;
+
+public class Ui {
+    private static final String DIVIDER = "---------------------------------------------------";
```
Good use of all uppercase letters to name constants

chengzihao123 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,138 @@
+     * @param task The task that was added
+     * @return A string containing the message that a task was added with the task description
+     */
+    public String displayAddedTask(Task task) {
```
Good to split a long line into several lines for greater readbility!

chengzihao123 reviewed (commented):
Overall good job! Just some nits to fix

rithanisk committed:
Add comment for TaskType enum

rithanisk committed:
Add getSymbol() method to all TaskTypes

rithanisk committed:
Update getTaskSymbol methods in Task subclasses

rithanisk committed:
Add saveTaskListToTextFile method

rithanisk committed:
Add getter methods for Deadline and Event class

rithanisk committed:
Complete saveTaskListToTextFile method

rithanisk committed:
Update main to save tasks to text file

rithanisk committed:
Add constructor to TaskList class

rithanisk committed:
Update main to check whether taskList file exists in data folder

rithanisk committed:
Add space in Deadline toString method

rithanisk committed:
Add textFileToArrayList method to read from existing taskList

rithanisk committed:
Merge branch-Level-7 with merge commit

rithanisk committed:
Add javadoc comments to Level-7 methods

rithanisk committed:
Add parseDateTimeOrDate method and format deadline as LocalDateTime

rithanisk committed:
Update getDeadline method to output deadline in specific format

rithanisk committed:
Add parseDateTimeOrDate method and format start and end as LocalDateTime

rithanisk committed:
Update getStart and getEnd to output start and end in specific format

rithanisk committed:
Define format for user inputted date and generated output date

rithanisk committed:
Update validateCommand to handle when date is entered with wrong format

rithanisk committed:
Add convertDate method to parse the dates from taskList txt file

rithanisk committed:
Add javadoc comments to Level-8 methods

rithanisk committed:
Merge branch-Level-8 with merge commit

rithanisk committed:
Add Storage class

rithanisk committed:
Update TaskList constructor

rithanisk committed:
Update Nebula class to follow OOP

rithanisk committed:
Add Command abstract class and subclasses

rithanisk committed:
Add new Command subclasses

rithanisk committed:
Add AddDeadlineCommand class

rithanisk committed:
Add AddEventCommand class

rithanisk committed:
Add AddTodoCommand class

rithanisk committed:
Add ByeCommand class

rithanisk committed:
Update Command abstract class

rithanisk committed:
Add DeleteCommand class

rithanisk committed:
Add ListCommand class

rithanisk committed:
Add MarkCommand class

rithanisk committed:
Add UnmarkCommand class

rithanisk committed:
Update Ui class methods to remove divider

rithanisk committed:
Add Parser class

rithanisk committed:
Refactor code in Neubla class

rithanisk committed:
Merge branch-A-MoreOOP with merge commit

rithanisk committed:
Organize classes into suitable java packages

rithanisk committed:
Merge branch-A-Packages with merge commit

rithanisk committed:
Merge add-gradle-support with merge commit

rithanisk committed:
Add EventTest file to test Event class methods

rithanisk committed:
Add createEvent_success method to EventTest

rithanisk committed:
Add assertEquals to test event without time in createEvent_success

rithanisk committed:
Add createEvent_failure method to EventTest

rithanisk committed:
Add assertEquals to test event with only end time in EventTest

rithanisk committed:
Add AddTodoCommandTest class to test AddTodoCommand class methods

rithanisk committed:
Add addTodo_success method to AddTodoCommandTest class

rithanisk committed:
Add ParserTest class to test parse method

rithanisk committed:
Merge branch-A-JUnit with merge commit

rithanisk committed:
Update dependencies in build.gradle file

rithanisk committed:
Merge branch-A-Jar with merge commit

rithanisk committed:
Add JavaDoc comments for Ui Parser and Nebula classes

rithanisk committed:
Add JavaDoc comments for TaskList class

rithanisk committed:
Add JavaDoc comments for Storage class

rithanisk committed:
Add JavaDoc comments to all the Command classes

rithanisk committed:
Added whitespace between lines in Command classes

rithanisk committed:
Reduce code line length to be less than 100 characters

rithanisk committed:
Add FindCommand class

rithanisk committed:
Update getTask method to return Task

rithanisk committed:
Update validateCommand method to handle find command

rithanisk committed:
Update Ui class with methods to print find command output

rithanisk committed:
Merge pull request #2 from rithanisk/branch-A-JavaDoc

Add JavaDoc comments to methods

rithanisk committed:
Merge branch 'master' into branch-A-CodingStandard

rithanisk committed:
Merge branch 'master' into branch-Level-9

rithanisk committed:
Merge pull request #3 from rithanisk/branch-A-CodingStandard

Format code to follow Java coding standard

rithanisk committed:
Merge branch 'master' into branch-Level-9

rithanisk committed:
Merge pull request #4 from rithanisk/branch-Level-9

Add find command functionality to Nebula chatbot

rithanisk committed:
Add basic Hello World GUI with JavaFX

rithanisk committed:
Add user input functionality in GUI

rithanisk committed:
Create basic GUI for Nebula chatbot

rithanisk committed:
Add automatic reize elements functionality to GUI

rithanisk committed:
Add css style to JavaFX files

rithanisk committed:
Add margins and padding to text message

rithanisk committed:
Remove divider from GUI

rithanisk committed:
Add assertions to Nebula class

rithanisk committed:
Add assert statements to Parser class

rithanisk committed:
Add assertion statements to classes in task package

rithanisk committed:
Fix spacing in Ui class

rithanisk committed:
Refactor validateCommand method in Parser class

rithanisk committed:
Delete comment in MainWindow

rithanisk committed:
Storage class: refactor task creation logic into separate methods

The Storage class has the task creation logic embedded within the
textFileToArrayList method.

This causes the code to be cluttered and harder to read.

Refactoring the task creation logic into separate methods allows
others looking at the textFileToArrayList method to easily
understand its function.

Let's refactor the task creation logic by extracting the following
methods:
- createTask: Creates Task based on taskSymbol
- createTodoTask: Creates Todo Task
- createDeadlineTask: Creates Deadline Task
- createEventTask: Creates Event Task

This approach simplifies the process of adding new Task types in
the future, enhances the overall code clarity, and improves readability

rithanisk committed:
Remove comments in FindCommand execute method

rithanisk committed:
Delete whitespace in Parser class

rithanisk committed:
Add semicolon to assert statement

rithanisk committed:
Merge pull request #5 from rithanisk/branch-A-Assertions

Add assert statements to code

rithanisk committed:
Merge branch 'master' into branch-A-CodeQuality

rithanisk committed:
Merge pull request #6 from rithanisk/branch-A-CodeQuality

Adhere to Java code quality standard

rithanisk committed:
Add HelpCommand class

rithanisk committed:
Update Parser to handle help command

rithanisk committed:
Add displayCommands method to Ui class for help command execution

rithanisk committed:
Change profile pictures of Nebula and user

rithanisk committed:
Change color scheme of chatbot

rithanisk committed:
Update ui for displayCommands method

rithanisk committed:
Add error handling logic for empty task description

rithanisk committed:
Improve code quality in Nebula and Parser classes

rithanisk committed:
Add JavaDoc comments to Ui class

rithanisk committed:
Add JavaDoc comments to Parser class

rithanisk committed:
Add JavaDoc comments for Ui classes

rithanisk committed:
Storage class: Add Javadoc comments to methods

The Storage class methods lack documentation, making it difficult for
other developers to understand their functionality and usage.

Let's add Javadoc comments to each method in the Storage class to
provide clear explanations of their purpose.

This documentation will enhance code readability and facilitate
maintenance by ensuring that all methods are well-explained.

rithanisk committed:
Replace 'Duke' with 'Nebula'

rithanisk committed:
Add JavaDoc comments to FindCommand and HelpCommand classes

rithanisk committed:
Add screenshot of chatbot GUI to docs folder

rithanisk committed:
Add user guide for Nebula chatbot

rithanisk committed:
Update product screenshot path
